src/shared/mod.rs

```diff
@@ -181,3 +183,21 @@ pub fn merge_nix_configs(
         .unwrap_or(&new_config)
         .to_owned()
 }
+
+pub async fn create_temp_netrc(dir: &Path, host: &str, token: &str) -> color_eyre::Result<PathBuf> {
```

Comment:
I wonder if this should take host as `&Url`, so that we can call `.host_str()` on it and be sure it's the right value always (I seem to remember you mentioning something about the netrc not working in your testing because it used the full URL instead of just the host string...?)

Reply:
I had that thought, too, and I'm going to take your having the same thought be the sign that it's time to change the function signature.